configuration/redis/metadata.yaml

```diff
@@ -27,20 +27,25 @@ metadata:
     example: "false"
     default: "false"
     type: bool
-  - name: maxRetries
+  - name: redisMaxRetries
```

Comment:
Note that this PR modifies redis metadata from `maxRetries` to `redisMaxRetries`, `maxRetryBackoff` to `redisMaxRetryInterval` and Adds another metadata `redisMinRetryInterval`. Renaming these metadata would be breaking changes in metadata, but since redis config store is alpha right now and moving to stable this release(1.11), this should be fine.

Also Since we are using internal redis component, most of the generic metadata options are supported like `dialTimeout`, `readTimeout`, poolSize` etc. Do we also need to included them in the metadata description and so in the associated docs? cc @berndverst

Reply:
You could, if you wanted, take the a similar path to the one taken by redis statestore and still read `maxRetries` and other legacy fields _but_ as you said, those are legacy. It's better to remove these legacy fields now and push for the new fields (which are consistent among all Redis component now) now.